scrypto-test/src/sdk/resource/bucket_factory.rs

```diff
@@ -170,7 +170,11 @@ impl BucketFactory {
                 // 1. All of one type.
                 // 2. This one type is the type of the non-fungible local ids.
                 let id_type = {
-                    let mut iter = non_fungibles.keys().map(|id| id.id_type());
+                    let mut iter = non_fungibles
+                        .keys()
+                        .map(|id| id.id_type())
+                        .collect::<HashSet<NonFungibleIdType>>()
+                        .into_iter();
```

Comment:
It doesn't matter in this case, but can we use IndexSet for consistency in the code base?